syntax = "proto3";

option go_package = "./wbparser";

package wbparser;

service WBParser {
  rpc GetPrice(BatchRequest) returns (BatchResponse);
}

message BatchRequest {
  repeated int32 nm_ids = 1;
}

message ProductResponse {
  int32 nm_id = 1;
  float price = 2;
  string name = 3;
  string error = 4;
}

message BatchResponse {
  repeated ProductResponse results = 1;
}